### PR TITLE
Care Controller: Remove evicted pods in Shoots control plane namespace on a Seed.

### PR DESCRIPTION
**Pull request by dkistner**
**What this PR does / why we need it**:
The care controller removes evicted pods in the Shoot control plane of a Seed.

**Which issue(s) this PR fixes**:
Fixes #

**Special notes for your reviewer**:
Checkout the PR and run the Gardener locally. Deploy a cluster and wait until it is created.
Then deploy a pod, which consumes plenty of memory e. g. via ``dd bs=8G if=/dev/zero of=/dev/null``, in the Shoot namespace on the Seed cluster.

**Release note**:

```improvement operator
There was an issue where evicted control plane pods in the shoot cluster namespace confused the health check into assuming the shoot cluster control plane is broken. Gardener will now delete these evicted pods prior to its actual health check and thereby avoid false alarms.
```

### CODE REVIEW

**Comment by vlerenc** [comment]
Thanks! I would rephrase the release note a bit (hide the implementation details and add the benefit):

> There was an issue where evicted control plane pods in the shoot cluster namespace confused the health check into assuming the shoot cluster control plane is broken. Gardener will now delete these evicted pods prior to its actual health check and thereby avoid false alarms.

**Comment by dkistner** [comment]
Thanks @vlerenc. I took it in the text :)

**Comment by dkistner** [comment]
@mvladev Could you please also have a look?

File: pkg/operation/botanist/garbage_collection.go [hunk 1]
```diff
@@ -62,7 +67,16 @@ func (b *Botanist) PerformGarbageCollectionSeed() error {
 // PerformGarbageCollectionShoot performs garbage collection in the kube-system namespace in the Shoot
 // cluster, i.e., it deletes evicted pods (mitigation for https://github.com/kubernetes/kubernetes/issues/55051).
 func (b *Botanist) PerformGarbageCollectionShoot() error {
-	podList, err := b.K8sShootClient.ListPods(metav1.NamespaceSystem, metav1.ListOptions{})
+	err := b.deleteEvictedPods(b.K8sShootClient, metav1.NamespaceSystem)
```
Comment (mvladev) [inline pkg/operation/botanist/garbage_collection.go]: This can be shortened to `return b.deleteEvictedPods(b.K8sShootClient, metav1.NamespaceSystem)`

**Review by mvladev — CHANGES_REQUESTED** [review]
Just a small change needed

**Review by mvladev — APPROVED** [review]
